src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -52,50 +67,70 @@ def from_remote_dandiset(
+            dandiset = client.get_dandiset(dandiset_id=dandiset_id, version_id=version_id)
 
-        dataset_description = get_bids_dataset_description(dandiset=dandiset)
+            dataset_description, _internal_messages = get_bids_dataset_description(dandiset=dandiset)
 
-        if limit is None:
-            assets = list(dandiset.get_assets())
-        else:
-            assets = [asset for counter, asset in enumerate(dandiset.get_assets()) if counter < limit]
+            if limit is None:
+                assets = list(dandiset.get_assets())
+            else:
+                assets = [asset for counter, asset in enumerate(dandiset.get_assets()) if counter < limit]
 
-        session_id_to_assets = collections.defaultdict(list)
-        for asset in assets:
-            asset_metadata = asset.get_raw_metadata()
+            session_id_to_assets = collections.defaultdict(list)
+            for asset in assets:
+                asset_metadata = asset.get_raw_metadata()
 
-            for session in asset_metadata.get("wasGeneratedBy", []):
-                if session.get("schemaKey", "") != "Session":
-                    continue
+                for session in asset_metadata.get("wasGeneratedBy", []):
+                    if session.get("schemaKey", "") != "Session":
+                        continue
 
-                session_id = session.get("identifier", "")
-                if session_id == "":
-                    continue
+                    session_id = session.get("identifier", "")
+                    if session_id == "":
+                        continue
 
-                session_id_to_assets[session_id].append(asset)
-        sorted_session_id_to_assets = dict(sorted(session_id_to_assets.items(), key=lambda item: item[0]))
+                    session_id_to_assets[session_id].append(asset)
+            sorted_session_id_to_assets = dict(sorted(session_id_to_assets.items(), key=lambda item: item[0]))
 
-        session_converters = [
-            SessionConverter(
-                session_id=session_id,
-                nwbfile_paths=[asset.get_content_url(follow_redirects=1, strip_query=True) for asset in assets],
-            )
-            for session_id, assets in sorted_session_id_to_assets.items()
-        ]
+            session_converters = [
+                SessionConverter(
+                    session_id=session_id,
+                    nwbfile_paths=[asset.get_content_url(follow_redirects=1, strip_query=True) for asset in assets],
+                )
+                for session_id, assets in sorted_session_id_to_assets.items()
+            ]
 
-        dataset_converter = cls(session_converters=session_converters, dataset_description=dataset_description)
+            dataset_converter = cls(session_converters=session_converters, dataset_description=dataset_description)
+            dataset_converter._internal_messages = _internal_messages
+            return dataset_converter
+        except:  # noqa
```

Comment:
```suggestion
        except Exception:  # noqa
```

The bare `except` would catch `SystemExit` and `KeyboardInterrupt` as well.

Reply:
What triggers `SystemExit`? I would not bother with capturing/reporting `KeyboardInterrupt`s